File: b/Python/PC.py
from Processor import Processor
from Disk import Disk
from RAM import RAM

class PC:
	# Konstruktor
	def __init__(self, name, pObj = Processor, dObj = Disk, rObj = RAM):
		self.__name = name
		self.__pObj = pObj
		self.__dObj = dObj
		self.__rObj = rObj
		self.__totalPrice = pObj.getPrice() + dObj.getPrice() + rObj.getPrice()

	# Setter & Getter
	def setName(self, name):
		self.__name = name

	def setProcessor(self, pObj = Processor):
		self.__pObj = pObj

	def setDisk(self, dObj = Disk):
		self.__dObj = dObj

	def setRAM(self, rObj = RAM):
		self.__rObj = rObj

	def setTotalPrice(self, pObj = Processor, dObj = Disk, rObj = RAM):
		self.__totalPrice = pObj.getPrice() + dObj.getPrice() + rObj.getPrice()
		
	def getName(self):
		return self.__name

	def getProcessor(self):
		return self.__pObj

	def getDisk(self):
		return self.__dObj

	def getRAM(self):
		return self.__rObj

	def getTotalPrice(self):
		return self.__totalPrice

	# Method
	def printPC(self):
		print("\n===== Prediksi Harga PC", self.__name, "=====\n")
		self.__pObj.printProcessor()
		self.__dObj.printDisk()
		self.__rObj.printRAM()
		print("-------------------------")
		print("Total Price : ", self.__totalPrice)

File: b/Java/PC.java
public class PC {
	// Atribut
	private Processor pObj;
	private Disk dObj;
	private RAM rObj;
	private String name;
	private int totalPrice;

	// Konstruktor
	public PC(String name, Processor pObj, Disk dObj, RAM rObj) {
		this.name = name;
		this.pObj = pObj;
		this.dObj = dObj;
		this.rObj = rObj;
		this.totalPrice = pObj.getPrice() + dObj.getPrice() + rObj.getPrice();
	}

	// Setter & Getter
	public void setName(String name) {
		this.name = name;
	}
	public void setProcessor(Processor pObj) {
		this.pObj = pObj;
	}
	public void setDisk(Disk dObj) {
		this.dObj = dObj;
	}
	public void setRAM(RAM rObj) {
		this.rObj = rObj;
	}
	public void setTotalPrice(Processor pObj, Disk dObj, RAM rObj) {
		this.totalPrice = pObj.getPrice() + dObj.getPrice() + rObj.getPrice();
	}

	public String getName() {
		return name;
	}
	public Processor getProcessor() {
		return pObj;
	}
	public Disk getDisk() {
		return dObj;
	}
	public RAM getRAM() {
		return rObj;
	}
	public int getTotalPrice() {
		return totalPrice;
	}

	// Method
	public void printPC() {
		System.out.println("\n===== Prediksi Harga PC " + name + " =====\n");
		pObj.printProcessor();
		dObj.printDisk();
		rObj.printRAM();
		System.out.println("-------------------------");
		System.out.println("Total Price : " + totalPrice);
	}
}